test/cmocka/src/math/arithmetic/base_10_logarithm.c

```diff
@@ -0,0 +1,105 @@
+	9.3855126936091011, 9.3931995222691196, 9.4007526601535094, 9.4081766782268659,
+	9.4154759169627091, 9.4226545015843630, 9.4297163562280168, 9.4366652161756566,
+	9.4435046406985137, 9.4502380233502628, 9.4568686022422757, 9.4633994693944423,
+	9.4698335793932600, 9.4761737574178788, 9.4824227066886628, 9.4885830153874373,
+	9.4946571630937573, 9.5006475267772306, 9.5065563863821918, 9.5123859300375031,
+	9.5181382589213257, 9.5238153918078847, 9.5294192693208828, 9.5349517579159713,
+	9.5404146536127215, 9.5458096854947918, 9.5511385189953373, 9.5564027589832818,
+	9.5616039526647825, 9.5667435923129869, 9.5718231178381536, 9.5768439193242560,
+	9.5818073388505756, 9.5867146733402073, 9.5915671761296206, 9.5963660590064990,
+	9.6011124940261787, 9.6058076152298781, 9.6104525202710605, 9.6150482719557271,
+	9.6195958997020572, 9.6240964009244330, 9.6285507423464711, 9.6329598611462810};
+
+/* testvector in Q32.0 */
+static const uint32_t uv[100] = {
+	1ULL,	 43383509ULL,	86767017ULL,   130150525ULL, 173534033ULL, 216917541ULL,
+	260301049ULL, 303684557ULL,  347068065ULL,  390451573ULL, 433835081ULL, 477218589ULL,
+	520602097ULL, 563985605ULL,  607369113ULL,  650752621ULL, 694136129ULL, 737519638ULL,
+	780903146ULL, 824286654ULL,  867670162ULL,  911053670ULL, 954437178ULL, 997820686ULL,
+	1041204194ULL, 1084587702ULL, 1127971210ULL, 1171354718ULL, 1214738226ULL, 1258121734ULL,
+	1301505242ULL, 1344888750ULL, 1388272258ULL, 1431655766ULL, 1475039274ULL, 1518422782ULL,
+	1561806290ULL, 1605189798ULL, 1648573306ULL, 1691956814ULL, 1735340322ULL, 1778723830ULL,
+	1822107338ULL, 1865490846ULL, 1908874354ULL, 1952257862ULL, 1995641370ULL, 2039024878ULL,
+	2082408386ULL, 2125791894ULL, 2169175403ULL, 2212558911ULL, 2255942419ULL, 2299325927ULL,
+	2342709435ULL, 2386092943ULL, 2429476451ULL, 2472859959ULL, 2516243467ULL, 2559626975ULL,
+	2603010483ULL, 2646393991ULL, 2689777499ULL, 2733161007ULL, 2776544515ULL, 2819928023ULL,
+	2863311531ULL, 2906695039ULL, 2950078547ULL, 2993462055ULL, 3036845563ULL, 3080229071ULL,
+	3123612579ULL, 3166996087ULL, 3210379595ULL, 3253763103ULL, 3297146611ULL, 3340530119ULL,
+	3383913627ULL, 3427297135ULL, 3470680643ULL, 3514064151ULL, 3557447659ULL, 3600831168ULL,
+	3644214676ULL, 3687598184ULL, 3730981692ULL, 3774365200ULL, 3817748708ULL, 3861132216ULL,
+	3904515724ULL, 3947899232ULL, 3991282740ULL, 4034666248ULL, 4078049756ULL, 4121433264ULL,
+	4164816772ULL, 4208200280ULL, 4251583788ULL, 4294967295ULL};
+
+static void test_math_arithmetic_base10log_fixed(void **state)
+{
+	(void)state;
+
+	uint32_t inpfxp[100];
+	double clogfxp;
+	double diff;
+	int i;
+
+	memcpy_s((void *)&inpfxp[0], sizeof(inpfxp), (void *)&uv[0], 100U * sizeof(uint32_t));
+	for (i = 0; i < ARRAY_SIZE(common_log10_ref_table); i++) {
+		clogfxp = log_10_int32(inpfxp[i]);
+		diff = fabs(common_log10_ref_table[i] - (double)clogfxp / (1 << 28));
+		printf("%20d|%20d|%20.10f|\n", i, inpfxp[i], clogfxp / (1 << 28));
```

Comment:
every time this function is called, it will print 100 lines?

Reply:
deleted data print.